### PR TITLE
feat(collections): filters

### PR DESCRIPTION
**Pull request by roiLeo**
I didn't find a way to remove empty collections from query yet

### PR type
- [x] Feature

### Before submitting Pull Request, please make sure:
- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted screenshot of demonstrated change in this PR

### What's new?
- [x] filters in /collections

### Screenshot
Screenshot 2021-11-25 at 09-17-39 Low minting fees and carbonless NFTs

### CODE REVIEW

File: src/components/rmrk/Gallery/Collections.vue [hunk 1]
```diff
@@ -137,8 +158,11 @@ export default class Collections extends Vue {
       result: this.handleResult,
       variables: () => {
         return {
+          orderBy: this.searchQuery.sortBy,
+          search: this.buildSearchParam(),
+          listed: this.searchQuery.listed ? [{price: { greaterThan: '0'}}] : [],
```
Comment (roiLeo) [inline src/components/rmrk/Gallery/Collections.vue:163]: 🤕

**Comment by yangwao** [comment]
> I didn't find a way to remove empty collections from query yet

Maybe @vikiival would have idea? would be amazing to close on this, collections are not great with empty collections so not wasting collectors attention, showing only those which can get would be interesting.

Like by default I would put something, show only collections where is at least 1 NFT for BUY

File: src/components/rmrk/Gallery/Collections.vue [hunk 2]
```diff
@@ -63,7 +65,8 @@ interface Image extends HTMLImageElement {
 type CollectionType = CollectionWithMeta;
 const components = {
   GalleryCardList: () => import('./GalleryCardList.vue'),
-  Search: () => import('./Search/SearchBar.vue'),
+  // Search: () => import('./Search/SearchBar.vue'),
```
Comment (vikiival) [inline src/components/rmrk/Gallery/Collections.vue:68]: remove unused import

File: src/components/rmrk/Gallery/Collections.vue (hunk 1 above)
Comment (vikiival) [reply src/components/rmrk/Gallery/Collections.vue:163]: looks legit

**Review by vikiival — APPROVED** [review]
Looks good to me

**Comment by vikiival** [comment]
> Maybe @vikiival would have idea? 

you can't filter on the nested entity
these are the fields you can filter

```graphql
input CollectionEntityFilter {
  version: StringFilter
  name: StringFilter
  max: IntFilter
  issuer: StringFilter
  symbol: StringFilter
  id: StringFilter
  metadata: StringFilter
  currentOwner: StringFilter
  events: JSONFilter
  blockNumber: BigFloatFilter
  createdAt: DatetimeFilter
  updatedAt: DatetimeFilter
  and: [CollectionEntityFilter!]
  or: [CollectionEntityFilter!]
  not: CollectionEntityFilter
}
```

the possible hack is to [count already minted NFTs in the collection entity](https://github.com/vikiival/magick/issues/11)

> Like by default I would put something, show only collections where is at least 1 NFT for BUY

It will skew the metrics

Screenshot 2021-11-25 at 20 27 29

**Comment by roiLeo** [comment]
> the possible hack is to [count already minted NFTs in the collection entity](https://github.com/vikiival/magick/issues/11)

We got the totalCount of nfts in collection but can't query/filter on > 0.
Does this work? https://github.com/vikiival/magick/pull/10 

> 
> It will skew the metrics

Buy now toggle will change metrics for sure. (as it filter nfts price > 0 in collection)

**Comment by vikiival** [comment]
> We got the totalCount of nfts in collection but can't query/filter on > 0.

Have you tried to run it on your local machine? 💻 
Is **think** that if we make the field as required it won't allow us to filter on top of that anyway

**Comment by yangwao** [comment]
> It will skew the metrics

Which metrics? Items? 
We can reformulate text to reflect what is shown as numbers. Let's try come up with a smart word for that

**Comment by yangwao** [comment]
Just chatted w @vikiival let's remove empty collection on UI till we figure out other solutions, just to have this working 😄

Lets do hacky solution for now, same with BUY stuff

**Comment by yangwao** [comment]
Bump, can we remove empty collections and let's merge this? 

image

**Comment by yangwao** [comment]
Bump, can we remove empty collections on UI and let's merge this @roiLeo ?  

image

**Comment by roiLeo** [comment]
> Bump, can we remove empty collections on UI and let's merge this @roiLeo ?

hotfix would be to remove card from row and it's not looking great
Screenshot 2021-11-30 at 15-15-37 Low minting fees and carbonless NFTs

I'm sure something can be done with graphql

**Comment by yangwao** [comment]
Yeah, but looking for empty collections is draining attention is worse than showing nothing 😄

Let's push it and we will "upgrade" it later, even now it's a huge step with functionality tho!

**Comment by yangwao** [comment]
WFM continue to #1357

**Pull request merged**